mmdeploy/codebase/mmdet/deploy/object_detection_model.py

```diff
@@ -604,6 +604,85 @@ class labels of shape [N, num_det].
+                 deploy_cfg: Union[str, Config], **kwargs):
+        assert backend == Backend.VACC, f'only supported vacc, but give \
+            {backend.value}'
+
+        super(VACCDetModel, self).__init__(backend, backend_files, device,
+                                           deploy_cfg, **kwargs)
+        # load cfg if necessary
+        model_cfg = load_config(model_cfg)[0]
+        self.model_cfg = model_cfg
+
+    def forward(self,
+                inputs: torch.Tensor,
+                data_samples: Optional[List[BaseDataElement]] = None,
+                mode: str = 'predict',
+                **kwargs) -> Any:
+        """The model forward.
+
+        Args:
+            inputs (torch.Tensor): The input tensors
+            data_samples (List[BaseDataElement], optional): The data samples.
+                Defaults to None.
+            mode (str, optional): forward mode, only support `predict`.
+        Returns:
+            Any: Model output.
+        """
+        assert mode == 'predict', 'Deploy model only allow mode=="predict".'
+        inputs = inputs.contiguous()
+        outputs = self.predict(inputs)
+        ret = self._get_bboxes(outputs[0], [i.metainfo for i in data_samples])
+        for i in range(len(ret)):
+            data_samples[i].pred_instances = ret[i]
+        return data_samples
+
+    def predict(self, imgs: Tensor, *args, **kwargs) -> List:
+        """Implement forward test.
+
+        Args:
+            imgs (Tensor): Input image(s) in [N x C x H x W] format.
+
+        Returns:
+            list[Tensor]: dets of shape [N, num_det, 5] and
+                class labels of shape [N, num_det].
+        """
+        outputs = self.wrapper({self.input_name: imgs})
+        outputs = [i for i in outputs.values()]
+        return outputs
+
+    def _get_bboxes(self, outputs, img_metas):
+        from mmdet.registry import MODELS
+        head_cfg = self.model_cfg._cfg_dict.model.bbox_head
+        head = MODELS.build(head_cfg)
+        if head_cfg.type == 'YOLOXHead':
+            divisor = round(len(outputs) / 3)
+            ret = head.predict_by_feat(
+                outputs[:divisor],
+                outputs[divisor:2 * divisor],
+                outputs[2 * divisor:], [dict(scale_factor=None)],
+                cfg=self.model_cfg._cfg_dict.model.test_cfg)
+        elif head_cfg.type == 'YOLOV3Head':
+            ret = head.predict_by_feat(
+                outputs, [dict(scale_factor=None)],
+                cfg=self.model_cfg._cfg_dict.model.test_cfg)
+        elif head_cfg.type in ('RetinaHead', 'SSDHead', 'FSAFHead'):
+            divisor = round(len(outputs) / 2)
+            ret = head.predict_by_feat(
+                outputs[:divisor],
+                outputs[divisor:],
+                img_metas=img_metas[0],
+                cfg=self.model_cfg._cfg_dict.model.test_cfg)
+        else:
+            raise NotImplementedError(f'{head_cfg.type} not supported yet.')
```

Comment:
I'm not sure if vacc backend apply the partition strategies the same as rknn. If not, may do adjustments.

Reply:
vacc backend follow the partition strategies from input to feature map output，maybe same as rknn. I tested yolov3 and can wrapper successed.

Reply:
You may keep yolov3-related codes if you only tested them.

Reply:
ok. I will delete yoloxhead and other except yolov3head